src/Service/HelloRetailRecommendationService.php

```diff
@@ -184,10 +184,23 @@ private function getProducts(array $productData, SalesChannelContext $context):
         $criteria->addAssociation('cover');
         $criteria->addAssociation('media');
         $criteria->addAssociation('seoUrls');
-        return $this->salesChannelRepository->search($criteria, $context)->getEntities();
+
+        $products = $this->salesChannelRepository->search($criteria, $context)->getEntities();
+
+        $hrData = new ProductModel(['results' => $productData]);
+        $hrStructs = $hrData->getStructs();
+
+        foreach ($products as $product) {
+            $productId = $product->getId();
+            if (isset($hrStructs[$productId])) {
+                $product->addExtension('hello-retail', $hrStructs[$productId]);
+            }
+        }
```

Comment:
```suggestion
        foreach ($hrData->getStructs() as $productId => $struct) {
            $product = $products->get($productId);
            if ($product) {
                $product->addExtension('hello-retail', $struct);
            }
        }
```

https://github.com/helloretail/shopware6-extension/blob/master/src/Models/ProductModel.php#L47
Alternativt, løb over structs hvis data mængden der er mindre.